controllers/restore_integration_test.go

```diff
@@ -0,0 +1,107 @@
+// +build integration
+
+package controllers
```

Comment:
Generaly, I prefer to use `_test` packages for tests. I deem this even more important for integration tests. I ensures, that only public things of the subject under test can be used.

Reply:
I find it a bit hard to generalize this idea. Sometimes I need to test package internal helper functions to keep test scope small without having to expose them. For me, same package visibility is fine as long as the file name suffix stays `_test.go`

Reply:
We've just discussed this in person, actually. I argued this as well. But I believe for an integration test it is a fair argument to make that only public APIs should be invoked.